### PR TITLE
Add sub command for completion and documentation generation

### LINKED ISSUE #480: Add additional sub commands for packagers
**What feature would you like to be added?**
Cobra supports automatic generation of [bash](https://github.com/spf13/cobra/blob/master/bash_completions.md) and zsh completions and [man](https://github.com/spf13/cobra/blob/master/doc/man_docs.md) pages. To use this features, additional command needs to be created.

**What are use cases of the feature?**
While not critical from the use of Kubeone, these features add to the usability and overall quality of it. Bash completions. Man pages are providing a universal method for UNIX users to view the documentation quickly while completions help user to avoid unnecessary typing and lookups when the functions of commands is already known.

### PR DESCRIPTION
**Pull request by eqrx**
**What this PR does / why we need it**:
See #480

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #480

**Special notes for your reviewer**:
Needed to upgrade cobra dependency to 0.0.4 because it has a helper method that simplifies argument parsing. Should the command be added to to documentation?

**Documentation**:
None

**Release note**:
```release-note
`completion` and `doc` added for shell complation and documentation generation
```

### CODE REVIEW

File: pkg/cmd/packaging.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	return cmd
+}
+
+func docCmd(rootCmd *cobra.Command) *cobra.Command {
+	var path string
+	var cmd = &cobra.Command{
+		Use:       "doc <man|md|rest|yaml>",
+		Short:     "Generates documentation",
+		Long:      "Generation can be generated as Man Pages, Markdown, ReStructured Text Docs or Yaml",
+		Example:   "kubeone completion bash",
```
Comment (kron4eg) [inline pkg/cmd/packaging.go]: should be `kubeone doc md`

Comment (eqrx) [reply pkg/cmd/packaging.go]: also changed it to `document`, seems more consistent for me for it to be a verb

File: pkg/cmd/packaging.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	"github.com/spf13/cobra/doc"
+)
+
+func completionCmd(rootCmd *cobra.Command) *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:   "completion <bash|zsh>",
+		Short: "Generates completion scripts for bash and zsh",
+		Long: `To load completion run into your current shell run
+
+. <(kubeone bashCompletion <shell>)
```
Comment (xmudrii) [inline pkg/cmd/packaging.go]: There's a typo in the description:

```suggestion
. <(kubeone completion <shell>)
```

File: pkg/cmd/packaging.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	return cmd
+}
+
+func documentCmd(rootCmd *cobra.Command) *cobra.Command {
+	var path string
+	var cmd = &cobra.Command{
+		Use:       "document <man|md|rest|yaml>",
+		Short:     "Generates documentation",
+		Long:      "Generation can be generated as Man Pages, Markdown, ReStructured Text Docs or Yaml",
+		Example:   "kubeone document mand",
```
Comment (xmudrii) [inline pkg/cmd/packaging.go]: This should probably be `man` or `md`

```suggestion
		Example:   "kubeone document man",
```

File: pkg/cmd/packaging.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	}
+	return cmd
+}
+
+func documentCmd(rootCmd *cobra.Command) *cobra.Command {
+	var path string
+	var cmd = &cobra.Command{
+		Use:       "document <man|md|rest|yaml>",
+		Short:     "Generates documentation",
+		Long:      "Generation can be generated as Man Pages, Markdown, ReStructured Text Docs or Yaml",
```
Comment (xmudrii) [inline pkg/cmd/packaging.go]: I would replace `Generation` with `Documentation` and maybe put everything in lower case:

```suggestion
		Long:      "Documentation can be generated as man pages, markdown, ReStructured Text Docs or yaml",
```

**Comment by xmudrii** [comment]
@eqrx Regarding the documentation question, a quick document explaining how to activate the completion and generate documentation would be nice. Maybe even just putting quick description in the README would work as well.

**Pull request merged**